Sources/Metrics/Metrics.swift

```diff
@@ -112,4 +115,44 @@ extension Timer {
 
         self.recordNanoseconds(nanoseconds.partialValue)
     }
+
+    #if compiler(>=6.0)
+    /// Convenience for measuring duration of a closure.
+    ///
+    /// - Parameters:
+    ///    - clock: The clock used for measuring the duration. Defaults to the continuous clock.
+    ///    - body: The closure to record the duration of.
+    @inlinable
+    @available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
```

Comment:
What's the recommendation for adopters who aren't yet on these OS versions as deployment targets?

Should we consider offering a variant if this method without the bumped OS requirements, which use pre-Clock way to measure time, for packages with older OS deployment targets? Or do we just recommend they write their own?